.github/workflows/lint.yml

```diff
@@ -15,6 +15,6 @@ jobs:
           stable: false
       - uses: actions/checkout@v2
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3
+        uses: golangci/golangci-lint-action@v5
```

Comment:
Update to newer version as I got the following linting error
```
func Result[T].{{func_name}} is unused (unused)
```
regarding [those functions](https://github.com/samber/mo/pull/42/files#diff-b67a0c563bd29e3d46d4cba43d67e8405b2d4a75f59b45d2a5eacaee606c6d44R216-R230)